src/py/flwr/server/superlink/fleet/grpc_rere/server_interceptor.py

```diff
@@ -172,3 +171,55 @@ def _generic_method_handler(
             request_deserializer=method_handler.request_deserializer,
             response_serializer=method_handler.response_serializer,
         )
+
+    def _verify_node_id(self, node_id: int, request: Request) -> bool:
+        if isinstance(request, CreateNodeRequest):
+            return False
+        if isinstance(request, PushTaskResRequest):
+            return request.task_res_list[0].task.consumer.node_id == node_id
```

Comment:
Should we check whether `task_res_list` is empty?

Reply:
Adding checks.